### PR TITLE
add statistic_broadcasts materialized view

### LINKED ISSUE #331: Statistic::Broadcast: Materialize database view
* **I'm submitting a ...**
  * user interface improvement

* **What is the current behavior?**
The API call at https://rundfunk-mitbestimmen.de/statistics takes about 1.9 seconds to finish.

* **What is the expected behavior?**
The API request should take less than a second.

* **What is the motivation / use case for changing the behavior?**
Improve performance.

* **What could be a possible solution? (not obligatory)**
Materialize the database view of `Statistic::Broadcast`. See instructions here: https://github.com/thoughtbot/scenic#what-about-materialized-views

Probably nothing more than `bin/rails g scenic:view statistic/broadcast --materialized` should do the trick.

### PR DESCRIPTION
**Pull request by ZmagoD**
fix #331

### CODE REVIEW

**Comment by roschaefer** [comment]
@ZmagoD run `bundle exec rubocop -a`

In the meantime I'll check the performance impact on write requests to the database.

File: backend/db/migrate/20171023194548_update_statistic_broadcasts_to_version_3.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class UpdateStatisticBroadcastsToVersion3 < ActiveRecord::Migration[5.0]
+  def changeđ
```
Comment (roschaefer) [inline backend/db/migrate/20171023194548_update_statistic_broadcasts_to_version_3.rb]: typo: `change`

**Comment by roschaefer** [comment]
OK the performance impact is very heavy: https://youtu.be/nFPp3NozFwM

We should try to use `concurrent: true` and update postgresql on the production server.

**Comment by roschaefer** [comment]
@ZmagoD you need to say `fix #331` or `close #331` to close the issue

File: backend/db/migrate/20171023194548_update_statistic_broadcasts_to_version_3.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class UpdateStatisticBroadcastsToVersion3 < ActiveRecord::Migration[5.0]
+  def change
+    drop_view :statistic_stations, revert_to_version: 1
```
Comment (ZmagoD) [inline backend/db/migrate/20171023194548_update_statistic_broadcasts_to_version_3.rb]: why we need to drop all views?

Comment (roschaefer) [reply backend/db/migrate/20171023194548_update_statistic_broadcasts_to_version_3.rb]: Other views depend on `statistic_broadcasts` and then postgres complaints about dependent views.

Comment (roschaefer) [reply backend/db/migrate/20171023194548_update_statistic_broadcasts_to_version_3.rb]: OK, wrong: When you have a view hierarchy you can use [replace view](https://github.com/thoughtbot/scenic#what-if-i-want-to-change-a-view-without-dropping-it). Unfortunately, that doesn't seem to work for the change non-materialized => materialized.

**Comment by ZmagoD** [comment]
@roschaefer can this be merged?

**Comment by roschaefer** [comment]
@ZmagoD hmm, I wouldn't merge this. As you can see in the video, the write access to the backend is very slow without concurrently updated views. I get an error on my machine with `concurrent: true` and have not further investigated how to set it up properly.

**Comment by ZmagoD** [comment]
Ok I will check it. In general what I read about materialized view you pay a cost on writing the data but is much faster for reading data so probably it depends how often the broadcast_statistics data is updated if make sense to have materialized views

**Comment by roschaefer** [comment]
@ZmagoD yep, exactly. But I wouldn't want to the HTTP request be blocked by updating the view. That's what `concurrent: true` does.

**Pull request merged**